### ISSUE TITLE
Conditions not working

### ISSUE THREAD

**Issue by koenigmartin** [awslabs/aws-service-catalog-puppet#484]
Hi Eamonn,
we ran into an issue testing conditions (puppet version: 0.154.0):
Following the Kitchen Sink Example we ran into an yaml error during validation and we had to alter the condition definition a bit:

```
conditions:
  IsRBIDev:
    "!Equals":
      - "rbi-dev"
      - "${Custom::env}"
```
The statement passed validation. When referencing the statement from a stack however we got the following error:
![conditions](https://user-images.githubusercontent.com/53942671/156121199-0c7e7070-a221-4cc7-b0e4-3417c5a2b47b.png)

We do get the same error when referencing a single condition as well as referencing multiple, so I dont think this is directly related to the multi org feature but a general issue:

```
stacks:
  ccoe-aws-sct-topic: 
    name: ccoe-aws-sct-topic
    condition: "IsRBIDev"
    ...
```

```
stacks:
  ccoe-aws-sct-sns-topic: 
    name: ccoe-aws-sct-topic
    condition: ["IsRBIDev"]
    ...
```
Best regards, Martin

**Comment by eamonnfaherty** [comment]
Would you be able to try the following condition instead please:

conditions:
  IsRBIDev:
    !Equals:
      - "rbi-dev"
      - "${Custom::env}"

I would need to check to confirm but I believe wrapping !Equals within double quotes would make it a string and may stop the custom yaml representer from working.

I have only tested this without the double strings and that is what is documented here: https://service-catalog-tools-workshop.com/managing-your-environments/gitops/conditions.html

**Comment by koenigmartin** [comment]
Using the annotation as provided in your post we run into the yaml error during validation:
![validation](https://user-images.githubusercontent.com/53942671/156150223-61d5025f-f069-4718-863b-609eb0b8b9d3.png)

**Comment by eamonnfaherty** [comment]
I will have a look at the validation error today

**Comment by eamonnfaherty** [comment]
I have had a look and it appears to be an error in your condition:

```
conditions:
  IsRBIDev:
    !Equals:
      - "rbi-dev"
      - "${Custom::env}"
```

should read

```
conditions:
  IsRBIDev:
    !Equals
      - "rbi-dev"
      - "${Custom::env}"
```

There should be no colon after !Equals

**Comment by koenigmartin** [comment]
```
conditions:
  IsRBIDev:
    !Equals
      - "rbi-dev"
      - "${Custom::env}"
```
also results in the following validation error:

![validation2](https://user-images.githubusercontent.com/53942671/156184336-1cd774b0-3973-4370-8532-c6d2f65960ef.png)

As stated in my first post, the validation fails if we use any other notation than the one given. 
There is no constructor for '!Equals' and it fails with your suggested one

**Comment by eamonnfaherty** [comment]
Could you please attach or email me your full manifest files please.

**Comment by eamonnfaherty** [comment]
I have just been able to reproduce this.  it works locally with pyyaml 5.4 but remotely with PyYAML-5.4-xx where xx is different it fails.

**Comment by eamonnfaherty** [comment]
Yamale (the library used for the schema parsing and validation) does not use the same yaml loader nor does it expose the option to chose a loader via its API.  I had to add in extra logic duplicated from its project to select the correct loader and then add the representer.

This fix is available in version 0.155.0

**Comment by koenigmartin** [comment]
I can confirm that this is fixed in 0.155.0.
Thank you very much for the fast responses and fixing the issue!
